File: PRODIGY_CS_03/Password complexity.c
#include <stdio.h>
#include <string.h>

int main() {
    char pwd[100];
    char str[20];
    char fb[5][100];
    int fbCnt;
    int len, u, l, d, s, met;

    while (1) {
        printf("Enter your password: ");
        scanf("%99s", pwd);

        len = strlen(pwd);
        u = l = d = s = 0;
        fbCnt = 0;

        // Check the length of the password
        if (len >= 8) {
            u = 1;
        } else {
            strcpy(fb[fbCnt++], "Password should be at least 8 characters long.");
        }

        // Check for uppercase, lowercase, digits, and special characters
        for (int i = 0; i < len; i++) {
            if (pwd[i] >= 'A' && pwd[i] <= 'Z') u = 1;
            if (pwd[i] >= 'a' && pwd[i] <= 'z') l = 1;
            if (pwd[i] >= '0' && pwd[i] <= '9') d = 1;
            if ((pwd[i] >= '!' && pwd[i] <= '/') || 
                (pwd[i] >= ':' && pwd[i] <= '@') || 
                (pwd[i] >= '[' && pwd[i] <= '`') || 
                (pwd[i] >= '{' && pwd[i] <= '~')) s = 1;
        }

        // Provide feedback for missing criteria
        if (!u) {
            strcpy(fb[fbCnt++], "Password should include at least one uppercase letter.");
        }
        if (!l) {
            strcpy(fb[fbCnt++], "Password should include at least one lowercase letter.");
        }
        if (!d) {
            strcpy(fb[fbCnt++], "Password should include at least one number.");
        }
        if (!s) {
            strcpy(fb[fbCnt++], "Password should include at least one special character.");
        }

        // Determine the strength of the password
        met = u + l + d + s;

        if (met == 4 && len >= 8) {
            strcpy(str, "Very Strong");
        } else if (met == 4) {
            strcpy(str, "Strong");
        } else if (met == 3) {
            strcpy(str, "Medium");
        } else if (met == 2) {
            strcpy(str, "Weak");
        } else {
            strcpy(str, "Very Weak");
        }

        printf("Password Strength: %s\n", str);
        if (fbCnt > 0) {
            printf("Feedback:\n");
            for (int i = 0; i < fbCnt; i++) {
                printf("- %s\n", fb[i]);
            }
        }

        if (strcmp(str, "Very Strong") == 0) {
            break;
        }
    }

    return 0;
}

File: PRODIGY_CS_03/Password complexity.py
import re

def f1(password):
    a = len(password) >= 8
    b = bool(re.search(r'[A-Z]', password))
    c = bool(re.search(r'[a-z]', password))
    d = bool(re.search(r'[0-9]', password))
    e = bool(re.search(r'[\W_]', password))
    f = a + b + c + d + e
    g = []
    if not a:
        g.append("Password should be at least 8 characters long.")
    if not b:
        g.append("Password should include at least one uppercase letter.")
    if not c:
        g.append("Password should include at least one lowercase letter.")
    if not d:
        g.append("Password should include at least one number.")
    if not e:
        g.append("Password should include at least one special character.")
    if f == 5:
        h = "Very Strong"
    elif f == 4:
        h = "Strong"
    elif f == 3:
        h = "Medium"
    elif f == 2:
        h = "Weak"
    else:
        h = "Very Weak"
    return h, g

while True:
    password = input("Enter your password: ")
    h, g = f1(password)
    print(f"Password Strength: {h}")
    if g:
        print("Feedback:")
        for i in g:
            print(f"- {i}")
    if h == "Very Strong":
        break

